packages/trie/src/db/flat/stackTrie.ts

```diff
@@ -0,0 +1,251 @@
+
+import { nibbleTypeToPackedBytes } from '../../util/index.js'
+import { bytesToNibbles, matchingNibbleLength } from '../../util/nibbles.js'
+
+import type { Nibbles } from '../../index.js'
+
+function addHexPrefix(key: Nibbles, terminator: boolean): Nibbles {
+  // odd
+  if (key.length % 2) {
+    key.unshift(1)
+  } else {
+    // even
+    key.unshift(0)
+    key.unshift(0)
+  }
+
+  if (terminator) {
+    key[0] += 2
+  }
+
+  return key
+}
+
+abstract class BaseNode {
+  abstract insert(key: Nibbles, value: Uint8Array): BaseNode
+  abstract raw(): any
+  serialize(): Uint8Array {
+    return RLP.encode(this.raw())
+  }
+  hash(): Uint8Array {
+    return keccak256(this.serialize())
+  }
+}
+
+export class EmptyNode extends BaseNode {
+  insert(key: Nibbles, value: Uint8Array): BaseNode {
+    //eslint-disable-next-line
+    return new STLeafNode(key, value)
+  }
+
+  raw(): any {
+    return null
+  }
+
+  serialize(): Uint8Array {
+    return new Uint8Array()
+  }
+
+  hash(): Uint8Array {
+    return KECCAK256_RLP
+  }
+}
+
+export class STLeafNode extends BaseNode {
+  _key: Nibbles
+  _value: Uint8Array
+
+  constructor(key: Nibbles, value: Uint8Array) {
+    super()
+    this._key = key
+    this._value = value
+  }
+
+  insert(key: Nibbles, value: Uint8Array): BaseNode {
+    const commonLen = matchingNibbleLength(this._key, key)
+    assert.ok(commonLen < this._key.length, 'replacing leaf is not supported')
+
+    const newLeaf = new STLeafNode(key.slice(commonLen + 1), value)
+
+    //eslint-disable-next-line
+    const branch = new STBranchNode()
+    branch.setChild(key[commonLen], newLeaf)
+    branch.setChild(this._key[commonLen], this)
+
+    let root: BaseNode = branch
+    // Need extension node for common path
+    if (commonLen > 0) {
+      //eslint-disable-next-line
+      root = new STExtensionNode(this._key.slice(0, commonLen), branch)
+    }
+
+    // Slice common part from current leaf's key
+    this._key = this._key.slice(commonLen + 1)
+
+    return root
+  }
+
+  raw(): [Uint8Array, Uint8Array] {
+    const val: Nibbles = this._key.slice(0)
+    const encodedKey = addHexPrefix(val, true) // add terminator value to end of key
+    return [nibbleTypeToPackedBytes(encodedKey), this._value]
+  }
+}
+
+export class STExtensionNode extends BaseNode {
+  _key: Nibbles
+  _child: BaseNode
+
+  constructor(key: Nibbles, child: BaseNode) {
+    super()
+    this._key = key
+    this._child = child
+  }
+
+  insert(key: Nibbles, value: Uint8Array): BaseNode {
+    const commonLen = matchingNibbleLength(this._key, key)
+    // If matching prefix, recurse into child node
+    if (this._key.length === commonLen) {
+      this._child = this._child.insert(key.slice(commonLen), value)
+      return this
+    }
+
+    // Otherwise we'll need a new leaf, a branch and possibly an extension
+    const newLeaf = new STLeafNode(key.slice(commonLen + 1), value)
+
+    //eslint-disable-next-line
```

Comment:
Is it not possible to just reorder the declarations so we don't need to have these eslint comments?

Reply:
Since all but the `BaseNode` type refer to each other, I wasn't able to remove all the eslint comments, but I was able to get it down to just two.